recipes/sleef/all/conanfile.py

```diff
@@ -24,75 +30,92 @@ class SleefConan(ConanFile):
         "fPIC": True,
     }
 
-    short_paths = True
-
-    exports_sources = "CMakeLists.txt"
-    generators = "cmake"
-    _cmake = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
-
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def validate(self):
         if self.settings.os == "Windows" and self.options.shared:
-            raise ConanInvalidConfiguration("shared sleef not supported on Windows, it produces runtime errors")
+            raise ConanInvalidConfiguration(
+                "shared sleef not supported on Windows, it produces runtime errors"
+            )
+        if self.settings.compiler == "apple-clang":
+            if cross_building(self):
+                # Fails with "No rule to make target `/bin/mkrename'"
+                # https://github.com/shibatch/sleef/issues/308
+                raise ConanInvalidConfiguration(f"{self.ref} does not support cross-building with apple-clang")
+            if Version(self.version) < "3.6" and self.settings.arch == "armv8":
+                # clang: error: the clang compiler does not support '-march=armv7-a'
+                # clang: warning: argument unused during compilation: '-mfpu=vfpv4' [-Wunused-command-line-argument]
+                # clang: warning: argument unused during compilation: '-arch arm64' [-Wunused-command-line-argument]
+                # clang: warning: argument unused during compilation: '-mmacosx-version-min=11.0' [-Wunused-command-line-argument]
+                raise ConanInvalidConfiguration(f"armv8 is not supported for {self.ref}")
```

Comment:
```suggestion
                raise ConanInvalidConfiguration(f"{self.ref} does not support Mac M1. Please, use {self.name} version >=3.6.")
```

Only a small detail, so people can know about which version supports M1 without reading the recipe.